### DISCUSSION TITLE
通过brew安装vim报错

### DISCUSSION THREAD

**Discussion by main1015** [Homebrew/discussions#5354] [Getting started]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.1.17
ORIGIN: https://mirrors.tuna.tsinghua.edu.cn/git/homebrew/brew.git
HEAD: 35746e0a6ba6c3c5cfe56d99f79d9ec9f52ee15f
Last commit: 7 months ago
Core tap JSON: 11 May 07:32 UTC
HOMEBREW_PREFIX: /usr/local
HOMEBREW_BOTTLE_DOMAIN: https://mirrors.cloud.tencent.com/homebrew-bottles
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 4
Homebrew Ruby: 2.6.10 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.10_1/bin/ruby
CPU: quad-core 64-bit haswell
Clang: 13.0.0 build 1300
Git: 2.32.0 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 8.1.2 => /usr/bin/curl
macOS: 11.7.10-x86_64
CLT: 13.2.0.0.1.1638488800
Xcode: N/A
```

### Output of `brew doctor`

```shell
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Some installed kegs have no formulae!
This means they were either deleted or installed manually.
You should find replacements for the following formulae:
  python-jinja
  python-pluggy
  python-hatchling
  python-pathspec
  python-setuptools-scm
  python-flit-core
  python-markupsafe

Warning: Some installed formulae are deprecated or disabled.
You should find replacements for the following formulae:
  openssl@1.1
  python-charset-normalizer
  python-idna
  python-requests
  python-trove-classifiers
  python-typing-extensions
  python-urllib3

Warning: You are using macOS 11.
We (and Apple) do not provide support for this old version.
It is expected behaviour that some formulae will fail to build in this old version.
It is expected behaviour that Homebrew will be buggy and slow.
Do not create any issues about this on Homebrew's GitHub repositories.
Do not create any issues even if you think this message is unrelated.
Any opened issues will be immediately closed without response.
Do not ask for help from Homebrew or its maintainers on social media.
You may ask for help in Homebrew's discussions but are unlikely to receive a response.
Try to figure out the problem yourself and submit a fix as a pull request.
We will review it but may or may not accept it.
```

### Description of issue

After I successfully installed through `brew install gnu-getopt`, I prompt me that some column commands like ` vim ` are not available, and when I run through the ` vimt. sh ` command, I prompt me:
`
dyld: Library not loaded: /usr/local/opt/ruby/lib/libruby.3.2.dylib
  Referenced from: /usr/local/bin/vim
  Reason: image not found
[1]    31233 abort      vim t.sh
`

I'll uninstall the old ` vim ` first
`
brew uninstall vim
`

Reinstall the ` vim ` again
`brew install vim`

I was prompted with the following error:
`
==> Fetching vim
==> Downloading https://raw.githubusercontent.com/Homebrew/homebrew-core/55e55c999586772fba17f6a6cfa544a39d380d27/Formula/v/vim.rb
Already downloaded: /Users/wangyu/Library/Caches/Homebrew/downloads/c65aaba944b14bcfdebb0a4fc23f4be7f9e5769b6768c423b97b76d3215bfe01--vim.rb
Error: vim: undefined method `throttle' for #<Livecheck:0x00007fd4d7f7e8e8>
Did you mean?  throw
`

**Comment by gromgit** [comment]
The `undefined method` error you see is likely because of this:

>```
>HOMEBREW_VERSION: 4.1.17
>ORIGIN: https://mirrors.tuna.tsinghua.edu.cn/git/homebrew/brew.git
>HEAD: 35746e0a6ba6c3c5cfe56d99f79d9ec9f52ee15f
>Last commit: 7 months ago
>```

Homebrew is already at v4.2.21. Either you haven't been running `brew update`, or the Tsinghua mirror you're using is broken.
